protocol.md

```diff
@@ -0,0 +1,136 @@
+# signing-spec Protocol
+
+March 03, 2021
+
+Version 0.1.0
+
+This document describes the protocol/algorithm for creating and verifying
+signing-spec signatures, independent of how they are transmitted or stored. For
+the recommended data structure, see [Envelope](envelope.md).
+
+## Signature Definition
+
+A signature is defined as:
+
+```none
+SIGNATURE = Sign(PAE(UTF8(PAYLOAD_TYPE), SERIALIZED_BODY))
+```
+
+Parameters:
+
+*   SERIALIZED_BODY is the byte sequence to be signed.
+
+*   PAYLOAD_TYPE is an authenticated URI indicating how to interpret
+    SERIALIZED_BODY. It encompasses the content type (JSON, Canonical-JSON,
+    CBOR, etc.), the purpose, and the schema version of the payload. This
+    obviates the need for the `_type` field within [in-toto]/[TUF] payloads.
+    This URI does not need to be resolved to a remote resource, nor does such a
+    resource need to be fetched. Examples: `https://in-toto.io/Link/v1.0`,
+    `https://in-toto.io/Layout/v1.0`,
+    `https://theupdateframework.com/Root/v1.0.5`.
+
+*   KEYID is an optional, unauthenticated hint indicating what key and algorithm
+    was used to sign the message. As with Sign(), details are agreed upon
+    out-of-band by the signer and verifier. It **MUST NOT** be used for security
+    decisions; it may only be used to narrow the selection of possible keys to
+    try.
+
+Functions:
+
+*   PAE() is the
+    [PASETO Pre-Authentication Encoding](https://github.com/paragonie/paseto/blob/master/docs/01-Protocol-Versions/Common.md#authentication-padding),
+    where parameters `type` and `body` are byte sequences:
+
+    ```none
+    PAE(type, body) := le64(2) || le64(len(type)) || type || le64(len(body)) || body
+    le64(n) := 64-bit little-endian encoding of `n`, where 0 <= n < 2^63
+    ```
+
+*   Sign() is an arbitrary digital signature format. Details are agreed upon
+    out-of-band by the signer and verifier. This specification places no
+    restriction on the signature algorithm or format.
+
+*   UTF8() is [UTF-8 encoding](https://tools.ietf.org/html/rfc3629),
+    transforming a unicode string to a byte sequence.
+
+## Protocol
+
+Out of band:
+
+-   Agree on a PAYLOAD_TYPE and cryptographic details, optionally including
+    KEYID.
+
+To sign:
+
+-   Serialize the message according to PAYLOAD_TYPE. Call the result
+    SERIALIZED_BODY.
+-   Sign PAE(UTF8(PAYLOAD_TYPE), SERIALIZED_BODY). Call the result SIGNATURE.
+-   Optionally, compute a KEYID.
+-   Encode and transmit SERIALIZED_BODY, PAYLOAD_TYPE, SIGNATURE, and KEYID.
+    -   In the recommended [JSON envelope](envelope.md), this is
+        `{"payload": "Base64(SERIALIZED_BODY)", "payloadType": "PAYLOAD_TYPE",
+        "signatures": [{"keyid": "KEYID", "sig": "Base64(SIGNATURE)"}]}`.
+
+To verify:
+
+-   Receive and decode SERIALIZED_BODY, PAYLOAD_TYPE, SIGNATURE, and KEYID.
+    Reject if decoding fails.
```

Comment:
Minor nit: Maybe also point to the envelope document here.

Reply:
Done. Please check the wording.